src/app/shared/menu/podcasts.component.ts

```diff
@@ -44,14 +43,13 @@ export class PodcastsComponent implements OnInit, OnDestroy {
   }
 
   ngOnDestroy() {
-    if (this.podcastFilterSubscription) { this.podcastFilterSubscription.unsubscribe(); }
-    if (this.intervalFilterSubscription) { this.intervalFilterSubscription.unsubscribe(); }
-    if (this.podcastsSubscription) { this.podcastsSubscription.unsubscribe(); }
+    if (this.filterSub) { this.filterSub.unsubscribe(); }
+    if (this.podcastsSub) { this.podcastsSub.unsubscribe(); }
   }
 
   onPodcastChange(val) {
     if (val && val.seriesId !== this.selectedPodcast.seriesId) {
-      this.router.navigate([val.seriesId, 'downloads', this.selectedInterval.key]);
+      this.router.navigate([val.seriesId, 'downloads', this.selectedChartType, this.selectedInterval.key]);
```

Comment:
General question: so is it the "ngrx" way to just `this.router.navigate` somewhere, and then that gets converted into an action at some point?  I guess I'm wondering how that compares to firing off a `this.store.dispatch` with some sort of "change route" action, and then that would do the navigation in a reducer or something.